egs/snips/v1/local/chain/tuning/run_e2e_tdnn_1a.sh

```diff
@@ -0,0 +1,280 @@
+      echo " ... not sure what to do.  Exiting."
+      exit 1;
+    fi
+  else
+    rm -rf $lang
+    cp -r data/lang $lang
+    silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+    nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+    # Use our special topology... note that later on may have to tune this
+    # topology.
+    local/gen_topo.pl 4 1 $nonsilphonelist $silphonelist >$lang/topo
+  fi
+fi
+
+if [ $stage -le 1 ]; then
+  echo "$0: Estimating a phone language model for the denominator graph..."
+  mkdir -p $tree_dir
+  id_sil=`cat data/lang/phones.txt | grep "SIL" | awk '{print $2}'`
+  id_word=`cat data/lang/phones.txt | grep "heysnips" | awk '{print $2}'`
+  id_freetext=`cat data/lang/phones.txt | grep "freetext" | awk '{print $2}'`
+  cat <<EOF > $tree_dir/phone_lm.txt
+0 1 $id_sil $id_sil
+0 5 $id_sil $id_sil
+1 2 $id_word $id_word
+2 3 $id_sil $id_sil
+1 4 $id_freetext $id_freetext
+4 5 $id_sil $id_sil
+3 2.09
+5 0.0
+EOF
+  fstcompile $tree_dir/phone_lm.txt $tree_dir/phone_lm.fst
+  fstdeterminizestar $tree_dir/phone_lm.fst $tree_dir/phone_lm.fst.tmp
+  mv $tree_dir/phone_lm.fst.tmp $tree_dir/phone_lm.fst
+  steps/nnet3/chain/e2e/prepare_e2e.sh --nj 30 --cmd "$train_cmd" \
+                                       data/${train_set}_hires $lang $tree_dir
+fi
+
+if [ $stage -le 2 ]; then
+  echo "$0: creating neural net configs using the xconfig parser";
+  num_targets=$(tree-info $tree_dir/tree | grep num-pdfs | awk '{print $2}')
+  affine_opts="l2-regularize=0.01 dropout-proportion=0.0 dropout-per-dim=true dropout-per-dim-continuous=true"
+  tdnnf_opts="l2-regularize=0.01 dropout-proportion=0.0 bypass-scale=0.66"
+  linear_opts="l2-regularize=0.01 orthonormal-constraint=-1.0"
+  prefinal_opts="l2-regularize=0.01"
+  output_opts="l2-regularize=0.002"
+
+  mkdir -p $dir/configs
+  cat <<EOF > $dir/configs/network.xconfig
+  input dim=40 name=input
+
+  # please note that it is important to have input layer with the name=input
+  # as the layer immediately preceding the fixed-affine-layer to enable
+  # the use of short notation for the descriptor
+
+  relu-batchnorm-dropout-layer name=tdnn1 input=Append(-2,-1,0,1,2) $affine_opts dim=$dim
+  tdnnf-layer name=tdnnf2 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf3 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf4 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf5 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf6 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf7 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf8 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=1
+  tdnnf-layer name=tdnnf9 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=0
+  tdnnf-layer name=tdnnf10 $tdnnf_opts dim=$dim bottleneck-dim=$bn_dim time-stride=3
```

Comment:
It's interesting that you have such a deep and narrow network.  Are you sure that you are getting a benefit from having so many layers?

Reply:
I remember I added layers one by one until it reached optimal. The left,right context now is both 41, which covers 0.8s duration. This optimal is obtained under a previous setup where negative examples are not segmented into pieces.  I can check if these many layers  are still optimal for the current setup.

Reply:
The left and right contexts are still both 41. Have you checked that they are still optimal?

I find that you have broken the negative samples into segments but the contexts
have not been changed.